bcda/models/models.go

```diff
@@ -304,10 +310,13 @@ type Resource struct {
 }
 
 type System struct {
-	ClientID   string `json:"client_id"`
-	SoftwareID string `json:"software_id"`
-	ClientName string `json:"client_name"`
-	ClientURI  string `json:"client_uri"`
+	gorm.Model
+	GroupID        string          `json:"group_id"`
+	ClientID       string          `json:"client_id"`
+	SoftwareID     string          `json:"software_id"`
+	ClientName     string          `json:"client_name"`
+	ClientURI      string          `json:"client_uri"`
+	EncryptionKeys []EncryptionKey `gorm:"foreignkey:SystemID" json:"encryption_keys"`
```

Comment:
should be []*EncryptionKey, with no annotation. See auth.Token model Token/ACO and bcda.models Job/JobKey